theano/gpuarray/opt.py

```diff
@@ -1990,43 +1991,104 @@ def local_inplace_gpu_solve(node):
 
 
 # Cholesky decomposition
-@register_opt('fast_compile')
-@op_lifter([slinalg.Cholesky])
-@register_opt2([theano.tensor.slinalg.Cholesky], 'fast_compile')
 def local_gpu_cholesky(op, context_name, inputs, outputs):
```

Comment:
Why did you undo that and did the work manually bellow the optimization? If this is truely needed, a commend should explain why. But I don't see why it is needed. The new version make understanding the code harder. If you need help with that section, I can help you this week, i'm back from GTC.

local_gpu_magma_cholesky should something similar to the 3 lines moved I think.